app/models/SearchHits.ts

```diff
@@ -14,53 +18,107 @@ export interface ServiceHit
   extends Omit<Service, "schedule" | "recurringSchedule" | "instructions">,
     BaseHit {
   type: "service";
-  instructions: string[];
-  phones: PhoneNumber[];
+  instructions: string[] | [];
+  phones: PhoneNumber[] | [];
   recurringSchedule: RecurringSchedule | null;
-  resource_schedule: ScheduleDay[];
-  schedule: ScheduleDay[];
+  resource_schedule: ScheduleDay[] | [];
+  schedule: ScheduleDay[] | [];
   service_id: number;
   service_of: string;
 }
-
 export interface OrganizationHit
   extends Omit<Organization, "schedule" | "recurringSchedule">,
     BaseHit {
   type: "resource";
   schedule: ScheduleDay[];
   recurringSchedule: RecurringSchedule | null;
 }
-
+export type SearchResultsResponse = AlogliaSearchResultsType<SearchHit>;
 export type SearchHit = ServiceHit | OrganizationHit;
+export type TransformedSearchHit = Hit<
+  SearchHit & {
+    recurringSchedule: RecurringSchedule | null;
+    resultIndexDisplay: string;
+    markerTag: string;
+    longDescription: string;
+    path: string;
+    headline: string;
+    resource_path: string;
+    geoLocPath: string;
+    phoneNumber: string | null;
+    websiteUrl: string | null;
+  }
+>;
+export interface SearchMapHitData
+  extends AlogliaSearchResultsType<TransformedSearchHit> {
+  hits: TransformedSearchHit[];
+}
 
-/**
- * Transform Algolia search hits such that each hit has a recurringSchedule that
- * uses the time helper classes.
- */
-export const addRecurringScheduleToSeachHits = (hits: SearchHit[]) => {
```

Comment:
This iterator now becomes a simple [getter](https://github.com/Exygy/askdarcel-web/pull/98/files?w=1#diff-904a52b3f75df52f7fd0be5e2975f9defd5ff30ea0324a0e98de2038e1b0565eR58). I wasn't concerned about performance but once I starting moving all computation to `TransformedSearchHit` this logic didn't have to modify any data in place.